src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+				performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				performCameraControl(Command.PAN_TILT_STOP, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+
+				// camera need 2s to apply new value
+				try {
+					Thread.sleep(2000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveLiveCameraInfo(stats);
+				populatePanTiltControls(stats, advancedControllableProperties);
+				break;
+			case HOME:
+				performCameraControl(Command.HASH.concat(Command.HOME_PRESET), value, controllableProperty, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				break;
+			case PT_SPEED:
+				cachedPanTiltControlSpeed = Float.parseFloat(value);
+				populatePanTiltControls(stats, advancedControllableProperties);
+				break;
+			default:
+				throw new IllegalStateException(String.format("preset control %s is not supported.", controllableProperty));
+		}
+		populatePresetControls(stats, advancedControllableProperties);
+	}
+
+	/**
+	 * This method is used to populate pan tilt control
+	 */
+	private String convertPanTiltPositionFromApiToUiValue(String panTiltData) {
+		if (panTiltData.length() == DeviceConstant.DEFAULT_PAN_POSITION.length() + DeviceConstant.DEFAULT_TILT_POSITION.length()) {
+			String currentPanValue = panTiltData.substring(0, DeviceConstant.DEFAULT_PAN_POSITION.length());
+			String currentTiltValue = panTiltData.substring(DeviceConstant.DEFAULT_PAN_POSITION.length());
+			String panPosition = String.valueOf((int) convertFromApiValueToUIValue(currentPanValue, DeviceConstant.MAX_PAN_POSITION_API_VALUE,
+					DeviceConstant.MIN_PAN_POSITION_API_VALUE, DeviceConstant.PAN_UI_API_CONVERT_FACTOR, DeviceConstant.MIN_PAN_POSITION_UI));
+			String tiltPosition = String.valueOf((int) convertFromApiValueToUIValue(currentTiltValue, DeviceConstant.MAX_TILT_POSITION_API_VALUE,
+					DeviceConstant.MIN_TILT_POSITION_API_VALUE, DeviceConstant.TILT_UI_API_CONVERT_FACTOR, DeviceConstant.MIN_TILT_POSITION_UI));
+			return String.format("Pan %s, tilt %s", panPosition, tiltPosition);
+		}
+		return DeviceConstant.NONE;
+	}
+
+	//--------------------------------------------------------------------------------------------------------------------------------
+	//endregion
+
+	//region zoom control
+	//--------------------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * This method is used to populate zoom control
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 */
+	private void populateZoomControls(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		String groupName = DevicesMetricGroup.ZOOM_CONTROL.getName() + DeviceConstant.HASH;
+
+		String zoomControlLabelStart = String.valueOf((int) DeviceConstant.MIN_ZOOM_UI_VALUE);
```

Comment:
Is the reason for String.valueof((int)0f) to have 0 instead of 0.0?

Reply:
because the response value is always in integer, so Nicolas wants to refactor to have 0 instead of 0.0